crux_core/src/capability.rs

```diff
@@ -1,14 +1,231 @@
+//! Capabilities provide a user-friendly API to request side-effects from the shell.
+//!
+//! Typically, capabilities provide I/O and host API access. Capabilities are external to the
+//! core Crux library. Some are part of the Crux core distribution, others are expected to be built by the
+//! community. Apps can also build single-use capabilities where necessary.
+//!
+//! # Example use
+//!
+//! A typical use of a capability would look like the following:
+//!
+//! ```rust,ignore
```

Comment:
these `ignore` attributes (that I peppered around the place a while back) should go. We could use `no_run`, but the code would need to compile so we'd need to add enough code around the example (using `#` to hide) to make it compile. Probs a job for another day though

Reply:
ha! Sorry just read the comments above! Glad we landed in the same place